src/plugins/intel_gna/tests/unit/gna_memory_alignment.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "any_copy.hpp"
+#include "backend/gna_limitations.hpp"
+#include "common/gna_target.hpp"
+#include "gna_data_types.hpp"
+#include "gna_plugin.hpp"
+#include "memory/gna_memory.hpp"
+#include "ngraph_functions/builders.hpp"
+
+using namespace InferenceEngine;
+using namespace ov::intel_gna::target;
+using namespace ov::intel_gna::limitations;
+namespace testing {
+
+struct MemAlignmentTestParams {
+    ExecutionMode execution_mode;
+    HWGeneration compile_target;
+    ngraph::Shape input_shape;
+    size_t mem_input_region_size;  // Expected memory size of the input region in bytes. For this specific model and
+                                   // when the value of input_shape_H is 1, the memory input region size can be
+                                   // calculated using below formula:
+                                   // mem_input_region_size = ALIGN8(input_shape_W)*inputPrecInBytes.
+                                   // Refer to GNAGraphCompiler::AffinePrimitive for more details.
+};
+
+std::vector<MemAlignmentTestParams> all_tests_params{
+    {ExecutionMode::SW_FP32, HWGeneration::UNDEFINED, {1, 2}, 32},
+    {ExecutionMode::SW_FP32, HWGeneration::UNDEFINED, {1, 8}, 32},
+    {ExecutionMode::SW_FP32, HWGeneration::UNDEFINED, {1, 9}, 64},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_0, {1, 2}, 64},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_0, {1, 32}, 64},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_0, {1, 33}, 128},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_5, {1, 2}, 64},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_5, {1, 32}, 64},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_5, {1, 33}, 128},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_6, {1, 2}, 16},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_6, {1, 8}, 16},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_6, {1, 9}, 32},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_3_6, {1, 33}, 80},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_4_0, {1, 2}, 16},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_4_0, {1, 8}, 16},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_4_0, {1, 9}, 32},
+    {ExecutionMode::SW_EXACT, HWGeneration::GNA_4_0, {1, 33}, 80},
+};
+
+class GNAPluginForMemoryAlignmentTest : public GNAPlugin {
+public:
+    GNAPluginForMemoryAlignmentTest(const std::map<std::string, std::string>& configMap) : GNAPlugin(configMap) {
+        if (gnadevice) {
+            gnamem.reset(new gna_memory_float(memory::GNAFloatAllocator{},
+                                              gnadevice->getMemAlignment(),
+                                              limitations::kMemoryPageSize));
+            graphCompiler.setGNAMemoryPtr(gnamem);
+            gnadevice.reset();
+        }
+    }
+
+    size_t get_memory_REGION_INPUTS_size() {
```

Comment:
it's nice to have getter be const. can we do this one?

Reply:
ok